gtsam/nonlinear/utilities.h

```diff
@@ -237,12 +237,12 @@ Values localToWorld(const Values& local, const Pose2& base,
       // if value is a Pose2, compose it with base pose
       Pose2 pose = local.at<Pose2>(key);
       world.insert(key, base.compose(pose));
-    } catch (std::exception e1) {
+    } catch (const std::exception& e1) {
       try {
         // if value is a Point2, transform it from base pose
         Point2 point = local.at<Point2>(key);
         world.insert(key, base.transformFrom(point));
-      } catch (std::exception e2) {
+      } catch (const std::exception& e2) {
```

Comment:
Can you add a debug output here if we are in debug mode?

Reply:
Absolutely

Reply:
Done